tests/conftest.py

```diff
@@ -96,12 +96,27 @@ def pytest_generate_tests(metafunc):
     'example_script' fixture.
     """
 
-    if "example_script" in metafunc.fixturenames:
-        this_file = pathlib.Path(__file__).parent.resolve()
-        example_path = this_file / pathlib.Path("../examples")
-        output_files = discover_python_scripts(example_path)
-        file_names = [str(file) for file in output_files]
-        metafunc.parametrize("example_script", output_files, ids=file_names)
+    if "example_script" not in metafunc.fixturenames:
+        return
+
+    # Required to only xfail a test if we are running a specific version of MI
+    # The example_script fixture is only used for integration tests, and if we are running integration tests we must
+    # have an accessible instance of Granta MI
+    mi_version = _get_mi_version()
+
+    this_file = pathlib.Path(__file__).parent.resolve()
+    example_path = this_file / pathlib.Path("../examples")
+    output_files = discover_python_scripts(example_path)
+    param_sets = [
```

Comment:
This is all a bit indirect, but I'm not sure of an alternative given how we dynamically configure these tests.

We'll probably need something like this in future though, at least with the mi_version check, given that the current development work will introduce examples that can only be executed against the latest release of the BAS API.

Reply:
I reverted these changes and instead deleted the broken sections of the example. This is because the docs build would still fail. We can revert this deletion once the underlying API issue is resolved.